devDocs/codingStandards.md

```diff
@@ -0,0 +1,86 @@
+## Code Style
+
+Python code style is enforced with the flake8 linter, see
+[`tests/lint/readme.md`](https://github.com/nvaccess/nvda/tree/master/tests/lint)
+for more information.
+
+### Encoding
+* Where Python files contain non-ASCII characters, they should be encoded in UTF-8.
+    * There should be no Unicode BOM at the start of the file, as this unfortunately breaks one of
+      the translation tools we use (`xgettext`).
+      Instead, include this as the first line of the file (only if the file contains non-ASCII
+      characters):
+        ```
+        # -*- coding: UTF-8 -*-
+        ```
+    * This coding comment must also be included if strings in the code (even strings that aren't
+      translatable) contain escape sequences that produce non-ASCII characters; e.g. `"\xff"`.
+      This is particularly relevant for braille display drivers.
+      This is due to a `gettext` bug; see
+      https://github.com/nvaccess/nvda/issues/2592#issuecomment-155299911.
+* Most files should contain `CRLF` line endings, as this is a Windows project and can't be used on
+  Unix-like operating systems.
+
+### Indentation
+* Indentation must be done with tabs (one per level), not spaces.
+* When splitting a single statement over multiple lines, just indent one or more additional levels.
+  Don't use vertical alignment; e.g. lining up with the bracket on the previous line.
+  - Be aware that this requires a new-line after an opening parenthesis/bracket/brace if you intend
+    to split the statement over multiple lines.
+  - For the parameter list of function definitions, double indent, this differentiates the
+    parameters and the body of the function.
+
+### Identifier Names
+* Use descriptive names
+  - name constants to avoid "magic numbers" and hint at intent or origin of the value.
+    Consider, what does this represent?
+* Functions, variables, properties, etc. use mixed case to separate words, starting with a lower
+  case letter; e.g. `speakText`.
+* Boolean functions or variables
+  - Prefer positive form of the language.
+    Avoid double negatives like `shouldNotDoSomething = False`
+  - Start with a "question word" to hint at their boolean nature. EG `shouldX`, `isX`, `hasX`
+* Classes should use mixed case to separate words, starting with an upper case letter;
+  - E.G. `BrailleHandler`.
+* Constants should be all upper case, separating words with underscores;
+  - E.G. `LANGS_WITH_CONJUNCT_CHARS`.
+* Event handlers are prefixed with "event_", subsequent words in capitals using snake case.
+  - E.G.: `event_ACTION` or `event_OBJECT_ACTION`.

```

Comment:
Is this true? event_gainFocus uses event_lowerCamelCase

Reply:
Hmm good point I'll update that.